### ISSUE TITLE
Create tool for removing org members

### ISSUE THREAD

**Issue by nikhita** [kubernetes/org#2052]
We intend to start doing a regular cleanup of inactive members (https://github.com/kubernetes/org/issues/2013, https://github.com/kubernetes/org/pull/2047).

Removing org members means removing them from the config files in this repo. Removing them manually is tedious so we need to create an automated tool that can help us do that.

https://github.com/kubernetes/org/pull/2048 has prior art in bash, we probably want the new tool to be written in Go.

**Comment by guineveresaenger** [comment]
/assign

I'd like to take this on!

**Comment by zouyee** [comment]
great

**Comment by guineveresaenger** [comment]
🙌

**Comment by ameukam** [comment]
@nikhita @mrbobbytables How the list of inactive members will be provided to the new tool ?

**Comment by dims** [comment]
@nikhita is this something we can turn into a GSoC project?

**Comment by mrbobbytables** [comment]
@dims this is probably too much of a little project. It's essentially to convert this bash script to go: https://github.com/kubernetes/org/pull/2458

**Comment by dims** [comment]
Ack @mrbobbytables

**Comment by spiffxp** [comment]
>It's essentially to convert this bash script to go: #2458

To me, that sounds like enough for...

/help
/kind cleanup
/priority important-longterm

**Comment by daemon1024** [comment]
/assign

I would like to work on this 😄

**Comment by pacoxu** [comment]
> @nikhita @mrbobbytables How the list of inactive members will be provided to the new tool?

+1 
The docs only says that we can check inactive member on devstats.cncf.io. But there is no simple query for that.

I was reminded to check inactive member per https://github.com/kubernetes/kubernetes/pull/104848#issuecomment-915541959

**Comment by mkumatag** [comment]
> devstats.cncf.io

I see devstats give us the [list of contributors](https://k8s.devstats.cncf.io/d/66/developer-activity-counts-by-companies?orgId=1&var-period_name=Last%202%20years&var-metric=contributions&var-repogroup_name=All&var-repo_name=kubernetes%2Fkubernetes&var-country_name=All&var-companies=All) based on the period like last 1year, 2 year etc.. based on the time limit we can get that list and negate with the total list to find the list of members we need to remove.

**Comment by mrbobbytables** [comment]
What we've done in the past is pull the stats for each release cycle which gives us an explicit period of time to track contributions, so if a person has not done anything in the past 5 releases (20 months now that releases are 4 months), we can safely remove.

**Comment by ameukam** [comment]
/remove-lifecycle stale
This is still needed.
/milestone v1.25

**Comment by ezzoueidi** [comment]
I can help with this. Looks like we need to add a function `RemoveInactiveMembers` in korg.go. We can fetch the needed list of inactive members from devstat [API](https://github.com/cncf/devstatscode/blob/master/API.md). 

Then to keep this automated, we can define a periodic prow job that runs every 2 years or so. 

What do you think folks? I don't want to submit a PR right away maybe someone has been working on that.

**Comment by blaisep** [comment]
I wanted to offer to adapt a github action that some of us built over in:
https://github.com/cucumber/action-retire-inactive-contributors/actions/runs/4361276398/jobs/7625018582

The description of the features is in:
https://github.com/cucumber/action-retire-inactive-contributors/tree/main/features

and a general diagram is at:
https://github.com/cucumber/action-retire-inactive-contributors#now-as-a-github-action

**Comment by Priyankasaggu11929** [comment]
Ppart of the requested functionality was added recently in (thanks @palnabarun):
- https://github.com/kubernetes/org/commit/8450b4ad44e640a3d345ba26ecbd570fb5760e65
- https://github.com/kubernetes/org/pull/4291

from: https://github.com/kubernetes/org/issues/2052#issuecomment-1444354344
> Then to keep this automated, we can define a periodic prow job that runs every 2 years or so.

@palnabarun, is this the only part left for implementation here?

**Comment by palnabarun** [comment]
I don't think we should run a Prow job for doing the audit. A lot of actions done in the process require human knowledge and intervention.

**Comment by Priyankasaggu11929** [comment]
> I don't think we should run a Prow job for doing the audit. A lot of actions done in the process require human knowledge and intervention.

I was thinking about having a Prow job, for *just* opening an issue with a `list of inactive org members` per year.

And then GH admins do the manual process for reviewing the list & actual removal.
